test/appender/test_appender.cpp

```diff
@@ -571,3 +571,187 @@ TEST_CASE("Test appending to different database files", "[appender]") {
+	duckdb::unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE OR REPLACE TABLE tbl (i INT DEFAULT 4, j INT, k INT DEFAULT 30)"));
+	Appender appender(con, "main", "tbl");
+
+	// Create a data chunk with all three columns filled.
+
+	DataChunk chunk_all_types;
+	const duckdb::vector<LogicalType> all_types = {LogicalType::INTEGER, LogicalType::INTEGER, LogicalType::INTEGER};
+	chunk_all_types.Initialize(*con.context, all_types);
+
+	setDataChunkInt32(chunk_all_types, 0, 0, 42);
+	setDataChunkInt32(chunk_all_types, 1, 0, 111);
+	setDataChunkInt32(chunk_all_types, 2, 0, 50);
+
+	chunk_all_types.SetCardinality(1);
+	appender.AppendDataChunk(chunk_all_types);
+
+	appender.AddColumn("j");
+	appender.AddColumn("i");
+
+	DataChunk chunk_j_i;
+	const duckdb::vector<LogicalType> types_j_i = {LogicalType::INTEGER, LogicalType::INTEGER};
+	chunk_j_i.Initialize(*con.context, types_j_i);
+
+	setDataChunkInt32(chunk_j_i, 0, 0, 111);
+	setDataChunkInt32(chunk_j_i, 1, 0, 42);
+
+	chunk_j_i.SetCardinality(1);
+	appender.AppendDataChunk(chunk_j_i);
+
+	appender.ClearColumns();
+	appender.AppendDataChunk(chunk_all_types);
+
+	appender.AddColumn("k");
+
+	DataChunk chunk_k;
+	const duckdb::vector<LogicalType> types_k = {LogicalType::INTEGER};
+	chunk_k.Initialize(*con.context, types_k);
+
+	setDataChunkInt32(chunk_k, 0, 0, 50);
+
+	chunk_k.SetCardinality(1);
+	appender.AppendDataChunk(chunk_k);
+	appender.Close();
+
+	result = con.Query("SELECT i, j, k FROM tbl");
+	REQUIRE(CHECK_COLUMN(result, 0, {42, 42, 42, 4}));
+	REQUIRE(CHECK_COLUMN(result, 2, {50, 30, 50, 50}));
+
+	// Test that column 1 is [111, 111, 111, NULL].
+	auto result_chunk = result->Fetch();
+	auto &null_result_vec = result_chunk->data[1];
+	UnifiedVectorFormat result_format;
+	null_result_vec.ToUnifiedFormat(result_chunk->size(), result_format);
+	REQUIRE(null_result_vec.GetValue(0).GetValue<int32_t>() == 111);
+	REQUIRE(null_result_vec.GetValue(1).GetValue<int32_t>() == 111);
+	REQUIRE(null_result_vec.GetValue(2).GetValue<int32_t>() == 111);
+	REQUIRE(!result_format.validity.RowIsValid(3));
+}
+
+TEST_CASE("Test edge cases for the active column configuration", "[appender]") {
+	duckdb::unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE OR REPLACE TABLE tbl (i INT DEFAULT 4, j INT, k INT DEFAULT 30, l AS (2 * j))"));
+	Appender appender(con, "main", "tbl");
+
+	appender.AddColumn("i");
+	appender.AddColumn("j");
+
+	bool failed;
+	// Cannot add columns that do not exist.
+	try {
+		failed = false;
+		appender.AddColumn("hello");
+	} catch (std::exception &ex) {
+		ErrorData error(ex);
+		REQUIRE(error.Message().find("the column must exist in the table") != std::string::npos);
+		failed = true;
+	}
+	REQUIRE(failed);
+
+	// Cannot add generated columns.
+	try {
+		failed = false;
+		appender.AddColumn("l");
```

Comment:
Can we verify that we can append to a column that appears **after** a generated column (i.e. add a column `m` to the table above and append there)?